### PR TITLE
Allow ignoring group writable bits

### PR DESCRIPTION
**Pull request by dentarg**
Resurrected https://github.com/fedora-ruby/gem-compare/pull/16 as it sounds like it makes sense to me (builds on #29 for now)

Tests ran here https://github.com/dentarg/gem-compare-1/actions/runs/2605538489

### CODE REVIEW

**Comment by strzibny** [comment]
Can you rebase please?

**Comment by dentarg** [comment]
@strzibny done, and I found (and fixed) a bug with the changes from #16

**Comment by dentarg** [comment]
> I found (and fixed) a bug with the changes from #16

I'll tack that back

```ruby
irb(main):043:0> sprintf "%o", 33188
=> "100644"
irb(main):044:0> 0100644
=> 33188
irb(main):045:0> 33188 == 0100644
=> true
```

**Comment by dentarg** [comment]
Feel free to review 

maybe we can ping the original contributor too

**Comment by ekohl** [comment]
I've been following the conversation since you pinged me on my PR. Thanks for completing it!

File: test/rubygems/comparator/test_file_list_comparator.rb [hunk 1]
```diff
@@ -9,10 +9,9 @@ def test_files_comparison
     assert_equal [], @report['files']['0.0.1->0.0.2']['deleted'].messages
     assert_equal [], @report['files']['0.0.1->0.0.2']['updated'].messages
     assert_equal "bin/lorem +3/-0", @report['files']['0.0.2->0.0.3']['added'].lines(1)
-    assert_equal "(!) Unexpected permissions: 100664", @report['files']['0.0.2->0.0.3']['added'].lines(2).strip
```
Comment (dentarg) [inline test/rubygems/comparator/test_file_list_comparator.rb:12]: It was incorrect to remove this line, added it back in #36

**Pull request merged**